service/src/test/java/org/databiosphere/workspacedataservice/dataimport/tdr/TdrManifestQuartzJobTest.java

```diff
@@ -235,4 +245,40 @@ void useWorkspaceIdFromCollection() throws JobExecutionException, IOException {
     // ASSERT
     verify(wsmDao).enumerateDataRepoSnapshotReferences(eq(workspaceId.id()), anyInt(), anyInt());
   }
+
+  @Test
+  @Tag(SLOW)
+  /* note: this functionality is control plane only.*/
+  void testSyncPermissions() throws JobExecutionException, IOException {
+    // ARRANGE
+    // set up ids
+    WorkspaceId workspaceId = WorkspaceId.of(UUID.randomUUID());
+    CollectionId collectionId = CollectionId.of(UUID.randomUUID());
+    UUID jobId = UUID.randomUUID();
+    // mock collection service to return this workspace id for this collection id
+    when(collectionService.getWorkspaceId(collectionId)).thenReturn(workspaceId);
+    // WSM should report no snapshots already linked to this workspace
+    // note that if enumerateDataRepoSnapshotReferences is called with the wrong workspaceId,
+    // this test will fail
+    when(wsmDao.enumerateDataRepoSnapshotReferences(eq(workspaceId.id()), anyInt(), anyInt()))
+        .thenReturn(new ResourceList());
+    // mock property that only gets enabled in application-control-plane.yml
+    when(dataImportProperties.isPermissionSync()).thenReturn(true);
+    // mock sam call
+    doNothing().when(samDao).addMemberPolicy(anyString(), any(), any(), anyString(), any());
+
+    // set up the job
+    TdrManifestQuartzJob tdrManifestQuartzJob =
+        testSupport.buildTdrManifestQuartzJob(workspaceId.id());
+    JobExecutionContext mockContext =
+        stubJobContext(
+            jobId, v2fManifestResource, collectionId.id(), /* shouldPermissionSync= */ true);
+
+    // ACT
+    // execute the job
+    tdrManifestQuartzJob.executeInternal(jobId, mockContext);
+
+    // ASSERT
+    verify(samDao, times(4)).addMemberPolicy(anyString(), any(), any(), anyString(), any());
```

Comment:
```suggestion
    ArgumentCaptor<String> roleCaptor = ArgumentCaptor.forClass(String.class);
    verify(samDao, times(4))
        .addMemberPolicy(anyString(), any(), any(), roleCaptor.capture(), any());
    assertThat(roleCaptor.getAllValues())
        .containsExactlyInAnyOrder("reader", "writer", "owner", "project-owner");
```

I'd suggest using an `ArgumentCaptor` here to check that the right roles were passed through.

Reply:
ccing @davidangb on this thread since he left a comment below- Id think the roles were the only significant param to check, right? This is a much cleaner way to do it so we don't have to call times(1) and check each param